### ISSUE TITLE
Node.js package installation fails on Node 13.0.1

### ISSUE THREAD

**Issue by joeduffy** [pulumi/pulumi#3434]
The Node.js Pulumi package fails to install on Node 13, due to gRPC build problems. This is now the "latest" version available. We've seen this on Windows and Linux — not sure about macOS. 

Here is a Windows build failure log:

```
Please choose a template: azure-typescript           A minimal Azure TypeScript Pulumi program
This command will walk you through creating a new Pulumi project.      

Enter a value or leave blank to accept the (default), and press <ENTER>.
Press ^C at any time to quit.

project name: (pulumi-demo2)
project description: (A minimal Azure TypeScript Pulumi program)
Created project 'pulumi-demo2'

Please enter your desired stack name.
To create a stack in an organization, use the format <org-name>/<stack-name> (e.g. `acmecorp/dev`).
stack name: (dev)
Created stack 'dev'

azure:environment: The Azure environment to use (`public`, `usgovernment`, `german`, `china`): (public)
azure:location: The Azure location to use: (WestUS) WestEU
Saved config

Installing dependencies...


> deasync@0.1.15 install C:\dev\git\personal\pulumi-demo2\node_modules\deasync
> node ./build.js


C:\dev\git\personal\pulumi-demo2\node_modules\deasync>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
  deasync.cc
  win_delay_load_hook.cc
     Creating library C:\dev\git\personal\pulumi-demo2\node_modules\deasync\build\Release\deasync.lib and object C:\dev\git\pe
  rsonal\pulumi-demo2\node_modules\deasync\build\Release\deasync.exp
  deasync.vcxproj -> C:\dev\git\personal\pulumi-demo2\node_modules\deasync\build\Release\\deasync.node
Installed in `C:\dev\git\personal\pulumi-demo2\node_modules\deasync\bin\win32-x64-node-13\deasync.node`

> grpc@1.21.1 install C:\dev\git\personal\pulumi-demo2\node_modules\grpc
> node-pre-gyp install --fallback-to-build --library=static_library

Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
  win_delay_load_hook.cc
  WINDOWS_BUILD_WARNING.vcxproj -> C:\dev\git\personal\pulumi-demo2\node_modules\grpc\build\Release\\WINDOWS_BUILD_WARNING.nod
  e
  err_data.c
  a_bitstr.c
  a_bool.c
  a_d2i_fp.c
C:\dev\git\personal\pulumi-demo2\node_modules\grpc\deps\grpc\third_party\boringssl\include\openssl\base.h(147,1): warning C400
5: 'OPENSSL_VERSION_NUMBER': macro redefinition (compiling source file ..\deps\grpc\src\boringssl\err_data.c) [C:\dev\git\pers
onal\pulumi-demo2\node_modules\grpc\build\boringssl.vcxproj]
C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\opensslv.h(42): message : see previous definition
 of 'OPENSSL_VERSION_NUMBER' (compiling source file ..\deps\grpc\src\boringssl\err_data.c) [C:\dev\git\personal\pulumi-demo2\n
ode_modules\grpc\build\boringssl.vcxproj]
C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\e_os2.h(171,1): warning C4005: 'OPENSSL_EXPORT':
macro redefinition (compiling source file ..\deps\grpc\src\boringssl\err_data.c) [C:\dev\git\personal\pulumi-demo2\node_module
s\grpc\build\boringssl.vcxproj]
C:\dev\git\personal\pulumi-demo2\node_modules\grpc\deps\grpc\third_party\boringssl\include\openssl\base.h(182): message : see
previous definition of 'OPENSSL_EXPORT' (compiling source file ..\deps\grpc\src\boringssl\err_data.c) [C:\dev\git\personal\pul
umi-demo2\node_modules\grpc\build\boringssl.vcxproj]
C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\ossl_typ.h(91,26): error C2371: 'EVP_MD': redefin
ition; different basic types (compiling source file ..\deps\grpc\src\boringssl\err_data.c) [C:\dev\git\personal\pulumi-demo2\n
ode_modules\grpc\build\boringssl.vcxproj]
C:\dev\git\personal\pulumi-demo2\node_modules\grpc\deps\grpc\third_party\boringssl\include\openssl\base.h(304): message : see
declaration of 'EVP_MD' (compiling source file ..\deps\grpc\src\boringssl\err_data.c) [C:\dev\git\personal\pulumi-demo2\node_m
odules\grpc\build\boringssl.vcxproj]
C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\ossl_typ.h(92,30): error C2371: 'EVP_MD_CTX': red
efinition; different basic types (compiling source file ..\deps\grpc\src\boringssl\err_data.c) [C:\dev\git\personal\pulumi-dem
o2\node_modules\grpc\build\boringssl.vcxproj]
C:\dev\git\personal\pulumi-demo2\node_modules\grpc\deps\grpc\third_party\boringssl\include\openssl\base.h(303): message : see
declaration of 'EVP_MD_CTX' (compiling source file ..\deps\grpc\src\boringssl\err_data.c) [C:\dev\git\personal\pulumi-demo2\no
de_modules\grpc\build\boringssl.vcxproj]
C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\ossl_typ.h(100,34): error C2371: 'EVP_ENCODE_CTX'
: redefinition; different basic types (compiling source file ..\deps\grpc\src\boringssl\err_data.c) [C:\dev\git\personal\pulum
i-demo2\node_modules\grpc\build\boringssl.vcxproj]
C:\dev\git\personal\pulumi-demo2\node_modules\grpc\deps\grpc\third_party\boringssl\include\openssl\base.h(308): message : see
declaration of 'EVP_ENCODE_CTX' (compiling source file ..\deps\grpc\src\boringssl\err_data.c) [C:\dev\git\personal\pulumi-demo
2\node_modules\grpc\build\boringssl.vcxproj]
C:\dev\git\personal\pulumi-demo2\node_modules\grpc\deps\grpc\third_party\boringssl\include\openssl\base.h(147,1): warning C400
5: 'OPENSSL_VERSION_NUMBER': macro redefinition (compiling source file ..\deps\grpc\third_party\boringssl\crypto\asn1\a_d2i_fp
.c) [C:\dev\git\personal\pulumi-demo2\node_modules\grpc\build\boringssl.vcxproj]
C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\opensslv.h(42): message : see previous definition
 of 'OPENSSL_VERSION_NUMBER' (compiling source file ..\deps\grpc\third_party\boringssl\crypto\asn1\a_d2i_fp.c) [C:\dev\git\per
sonal\pulumi-demo2\node_modules\grpc\build\boringssl.vcxproj]
C:\dev\git\personal\pulumi-demo2\node_modules\grpc\deps\grpc\third_party\boringssl\include\openssl\base.h(147,1): warning C400
5: 'OPENSSL_VERSION_NUMBER': macro redefinition (compiling source file ..\deps\grpc\third_party\boringssl\crypto\asn1\a_bool.c
) [C:\dev\git\personal\pulumi-demo2\node_modules\grpc\build\boringssl.vcxproj]
C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\opensslv.h(42): message : see previous definition
 of 'OPENSSL_VERSION_NUMBER' (compiling source file ..\deps\grpc\third_party\boringssl\crypto\asn1\a_bool.c) [C:\dev\git\perso
nal\pulumi-demo2\node_modules\grpc\build\boringssl.vcxproj]
C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\e_os2.h(171): message : see previous definition o
f 'OPENSSL_EXPORT' (compiling source file ..\deps\grpc\third_party\boringssl\crypto\asn1\a_bool.c) [C:\dev\git\personal\pulumi
-demo2\node_modules\grpc\build\boringssl.vcxproj]
C:\dev\git\personal\pulumi-demo2\node_modules\grpc\deps\grpc\third_party\boringssl\include\openssl\base.h(231,13): error C2371
: 'CRYPTO_THREADID': redefinition; different basic types (compiling source file ..\deps\grpc\third_party\boringssl\crypto\asn1
\a_d2i_fp.c)C:\dev\git\personal\pulumi-demo2\node_modules\grpc\deps\grpc\third_party\boringssl\include\openssl\base.h(231,13):
 error C2371: 'CRYPTO_THREADID': redefinition; different basic types (compiling source file ..\deps\grpc\third_party\boringssl
\crypto\asn1\a_bool.c)C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\crypto.h(231,3): error C237
1: 'CRYPTO_THREADID': redefinition; different basic types (compiling source file ..\deps\grpc\src\boringssl\err_data.c)C:\dev\
git\personal\pulumi-demo2\node_modules\grpc\deps\grpc\third_party\boringssl\include\openssl\base.h(147,1): warning C4005: 'OPE
NSSL_VERSION_NUMBER': macro redefinition (compiling source file ..\deps\grpc\third_party\boringssl\crypto\asn1\a_bitstr.c) [C:
\dev\git\personal\pulumi-demo2\node_modules\grpc\build\boringssl.vcxproj]
C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\opensslv.h(42): message : see previous definition
 of 'OPENSSL_VERSION_NUMBER' (compiling source file ..\deps\grpc\third_party\boringssl\crypto\asn1\a_bitstr.c) [C:\dev\git\per
sonal\pulumi-demo2\node_modules\grpc\build\boringssl.vcxproj]
C:\Users\MarcDuiker\AppData\Local\node-gyp\Cache\13.0.1\include\node\openssl\crypto.h(231): message : see declaration of 'CRYP
TO_THREADID' (compiling source file ..\deps\grpc\third_party\boringssl\crypto\asn1\a_bool.c)C:\Users\MarcDuiker\AppData\Local\
Attempting to cancel the build...
```

**Comment by lukehoban** [comment]
See https://github.com/grpc/grpc-node/issues/1086.

**Comment by jmck-contessa** [comment]
grpc/grpc-node#1086 is closed and grpc-node 1.24.2 has been released with support for Node 13.

https://github.com/grpc/grpc-node/issues/1086#issuecomment-548891990

**Comment by dibble-james** [comment]
A little workaround that has worked for me whilst 1.24.2 filters it's way through is to fudge the Pulumi GRPC dependency using an `npm-shrikwrap.json` next to your package.json.  Something like this:
```
{
    "dependencies": {
        "@pulumi/pulumi": {
            "version": "^1.0.0",
            "from": "@pulumi/pulumi@1.0.0",
            "dependencies": {
                "grpc": {
                    "version": "1.24.2",
                    "from": "grpc@1.21.1"
                }
            }
        }
    }
}
```

**Comment by pgavlin** [comment]
I have verified that this is fixed in the `"dev"` tag of `@pulumi/pulumi`. The fix will go out tomorrow with the 1.6.0 release.
